### PR TITLE
Idisp indexes

### PR DESCRIPTION
**Pull request by juimonen**
demux topologies won't load without this change in apl/up2

### CODE REVIEW

File: tools/topology/sof-hda-generic-idisp.m4 [hunk 1]
```diff
@@ -104,9 +104,9 @@ PCM_PLAYBACK_ADD(HDMI3, 3, PIPELINE_PCM_4)
 #
 
 # 3 HDMI/DP outputs (ID: 1,2,3)
-DAI_CONFIG(HDA, 0, 1, iDisp1)
-DAI_CONFIG(HDA, 1, 2, iDisp2)
-DAI_CONFIG(HDA, 2, 3, iDisp3)
+DAI_CONFIG(HDA, 0, 3, iDisp1)
+DAI_CONFIG(HDA, 1, 4, iDisp2)
+DAI_CONFIG(HDA, 2, 5, iDisp3)
```
Comment (plbossart) [inline tools/topology/sof-hda-generic-idisp.m4]: what does this have to do with demux? Or this is to align with recent changes on the hdaudio machine driver?

Comment (plbossart) [reply tools/topology/sof-hda-generic-idisp.m4]: also the second index is the dailink_index, and I don't see it having changed so far:

````c
struct snd_soc_dai_link skl_hda_be_dai_links[HDA_DSP_MAX_BE_DAI_LINKS] = {
	/* Back End DAI links */
	{
		.name = "iDisp1",
		.id = 1,
		.dpcm_playback = 1,
		.no_pcm = 1,
		SND_SOC_DAILINK_REG(idisp1_cpu, idisp1_codec, platform),
	},
	{
		.name = "iDisp2",
		.id = 2,
		.dpcm_playback = 1,
		.no_pcm = 1,
		SND_SOC_DAILINK_REG(idisp2_cpu, idisp2_codec, platform),
	},
	{
		.name = "iDisp3",
		.id = 3,
		.dpcm_playback = 1,
		.no_pcm = 1,
		SND_SOC_DAILINK_REG(idisp3_cpu, idisp3_codec, platform),
	},
````

**Comment by juimonen** [comment]
@kv2019i is this correct?

**Comment by kv2019i** [comment]
@juimonen wrote (and @plbossart asked):
> @kv2019i is this correct?

This is related to @lyakh changed https://github.com/thesofproject/sof/pull/2301/commits/ad7b1d20029eb01b978d94becd9d48566880cf5a that was recently merged. 

Kernel driver is now changes to use to new ids, so all topologies need to match.

**Comment by juimonen** [comment]
@kv2019i but I should not change this in generic-hda, which was the other commit in this pr?

**Comment by kv2019i** [comment]
> @kv2019i but I should not change this in generic-hda, which was the other commit in this pr?

Aa, sorry -- no, the generic-hda should not be changed, the ids were only changed for up2-pcm512.

**Comment by juimonen** [comment]
@kv2019i @plbossart ok my mistake. Removed hda-generic from this PR, now only includes the demux-apl prt.

File: tools/topology/sof-apl-demux-pcm512x.m4 [hunk 2]
```diff
@@ -152,8 +152,8 @@ DAI_CONFIG(SSP, 5, 0, SSP5-Codec,
 		SSP_CONFIG_DATA(SSP, 5, 24)))
 
 # 3 HDMI/DP outputs (ID: 1,2,3)
-DAI_CONFIG(HDA, 0, 1, iDisp1)
-DAI_CONFIG(HDA, 1, 2, iDisp2)
-DAI_CONFIG(HDA, 2, 3, iDisp3)
+DAI_CONFIG(HDA, 0, 3, iDisp1)
```
Comment (plbossart) [inline tools/topology/sof-apl-demux-pcm512x.m4]: the comment above is incorrect

Comment (juimonen) [reply tools/topology/sof-apl-demux-pcm512x.m4]: ok fixed now.

File: tools/topology/sof-apl-demux-pcm512x.m4 [hunk 3]
```diff
@@ -151,9 +151,9 @@ DAI_CONFIG(SSP, 5, 0, SSP5-Codec,
...
 		SSP_CONFIG_DATA(SSP, 5, 24)))
 
-# 3 HDMI/DP outputs (ID: 1,2,3)
-DAI_CONFIG(HDA, 0, 1, iDisp1)
-DAI_CONFIG(HDA, 1, 2, iDisp2)
-DAI_CONFIG(HDA, 2, 3, iDisp3)
+# 3 HDMI/DP outputs (ID: 3,4,5)
+DAI_CONFIG(HDA, 0, 3, iDisp1)
+DAI_CONFIG(HDA, 1, 4, iDisp2)
+DAI_CONFIG(HDA, 2, 5, iDisp3)
```
Comment (plbossart) [inline tools/topology/sof-apl-demux-pcm512x.m4]: ok the indices look fine, but I am not sure I understand this line


````
# currently this dai is here as "virtual" capture backend
W_DAI_IN(SSP, 5, SSP5-Codec, s24le, 3, 0)
````

what is the purpose of this line?

What happens if we truly want to use SSP5 for capture?

Comment (juimonen) [reply tools/topology/sof-apl-demux-pcm512x.m4]: @plbossart I have to test and think that a bit. Frankly I was just now trying to load the demux component for testing my topology work and loading stopped because of wrong idisp indexes. IOW I didn't actually test does the topology work, with this it at least loads.

**Comment by juimonen** [comment]
anything blocking of merging this simple PR?

**Comment by plbossart** [comment]
> anything blocking of merging this simple PR?

" I didn't actually test does the topology work, with this it at least loads."

Shouldn't we have functional topologies in the SOF tree?

**Comment by juimonen** [comment]
@plbossart well ok let's say this is an improvement to the current status. But yeah, no traction so let's close it. I can push it later if someone needs it. Maybe then remove the apl-demux conf altogether as it really doesn't load atm?
